src/flag_gems/ops/unique.py

```diff
@@ -0,0 +1,846 @@
+import torch
+import triton
+import triton.language as tl
+
+from flag_gems.utils.libentry import libentry
+
+
+@libentry()
+@triton.jit
+def simple_unique_flat_kernel(
+    sorted_data_ptr: tl.tensor,
+    sorted_indices_ptr: tl.tensor,  # in
+    data_out_ptr: tl.tensor,
+    inverse_indices_ptr: tl.tensor,
+    idx_ptr: tl.tensor,
+    unique_size_ptr: tl.tensor,  # out
+    return_inverse: tl.constexpr,
+    return_counts: tl.constexpr,
+    num_tasks: int,
+    tile_size: tl.constexpr,
+):
+    i0 = tl.arange(0, tile_size)
+    mask = i0 < num_tasks
+
+    # load
+    a = tl.load(sorted_data_ptr + i0, mask=mask)
+    i0_prev = tl.where(i0 > 0, i0 - 1, 0)
+    b = tl.load(sorted_data_ptr + i0_prev, mask=mask)
+
+    # ne & cumsum
+    ne_result = tl.where(i0 > 0, a != b, 0)
+    cumsum = tl.cumsum(ne_result)
+
+    # unique_size
+    unique_size = tl.sum(ne_result)
+    tl.store(unique_size_ptr, unique_size)
```

Comment:
Simply storing the last cumsum may do the work.

```suggestion
    tl.store(unique_size_ptr + i0*0, cumsum, mask=i0 == (tile_size-1))
```